prow/cmd/webhook-server/main.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+	"k8s.io/test-infra/experiment/clustersecretbackup/secretmanager"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+const (
+	secretID  = "prowjob-webhook-ca-cert"
+	caCert = "ca-cert"
+	caPrivKey = "ca-priv-key"
+	caBundle = "ca-bundle"
+	gcpSecretError = "failed to access secret version"
+)
+
+type ClientInterface interface {
+	CreateSecret(ctx context.Context, secretID string) (*secretmanagerpb.Secret, error)
+	AddSecretVersion(ctx context.Context, secretName string, payload []byte) error
+	GetSecretValue(ctx context.Context, secretName, versionName string) ([]byte, error)
+}
+
+func main() {
+	logrusutil.ComponentInit()
+	logrus.SetLevel(logrus.DebugLevel)
+	p := flag.String("project-id", 
+	"colew-test", "Project ID for storing GCP Secrets")
+	e := flag.Int("expiry-years", 30, "CA certificate expiry in years")
+	dns := flagutil.NewStrings("validation-webhook-service", "validation-webhook-service.default", "validation-webhook-service.default.svc")
+	flag.Var(&dns, "dns", "DNS Names CA-Cert config")
+	flag.Parse()
+	exp := *e
+	projectId := *p
+	if len(projectId) == 0 {
+		logrus.Fatal("project-id flag not supplied")
+	}
+	client, err := secretmanager.NewClient(projectId, false)
+	if err != nil {
+		logrus.WithError(err).Fatal("Unable to create secret manager client")
+	}
+	ctx := context.Background()
+	var secretUnavailable bool
```

Comment:
I don't think you even need this boolean. This can be simplified if you

```
if err != nil && strings.Contains(err.Error(), gcpSecretError) {
     // Secret does not exist yet, Create Secret
} else if err != nil {
  // return error
}
...
```